pkg/capr/configserver/agent.go

```diff
@@ -0,0 +1,115 @@
+package configserver
+
+import (
+	"bytes"
+	"net/http"
+	"sort"
+
+	"github.com/gorilla/mux"
+	"github.com/rancher/rancher/pkg/capr"
+	"github.com/rancher/rancher/pkg/controllers/dashboard/clusterindex"
+	"github.com/rancher/rancher/pkg/settings"
+	"github.com/rancher/rancher/pkg/systemtemplate"
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	capi "sigs.k8s.io/cluster-api/api/v1beta1"
+)
+
+// connectClusterAgentYAML renders a cattle-cluster-agent manifest for the given bearer token
+func (r *RKE2ConfigServer) connectClusterAgentYAML(rw http.ResponseWriter, req *http.Request) {
+	token := mux.Vars(req)["token"]
+	if token == "" {
+		http.Error(rw, "unauthorized", http.StatusUnauthorized)
+		return
+	}
+
+	tokens, err := r.clusterTokenCache.GetByIndex(tokenIndex, token)
```

Comment:
Can we rename the variable to minimize confusion?
```suggestion
	clusterTokens, err := r.clusterTokenCache.GetByIndex(tokenIndex, token)
```